jupyter_ydoc/ybasedoc.py

```diff
@@ -97,8 +98,7 @@ def dirty(self, value: bool) -> None:
         :param value: Whether the document is clean or dirty.
         :type value: bool
         """
-        with self._ydoc.begin_transaction() as t:
-            self._ystate.set(t, "dirty", value)
+        self._ystate["dirty"] = value
```

Comment:
Some much better API :heart:

Reply:
Yes I should write more documentation for pycrdt. Here an implicit transaction is created. Nested transactions are also possible:
```py
def foo(doc):
    with doc.transaction()
        ...

with doc.transaction():
    ...
    foo(doc)
```
And the outer-most transaction is used, while with Ypy this would raise an error.